packages/analytics-js/src/components/userSessionManager/UserSessionManager.ts

```diff
@@ -308,10 +311,11 @@ class UserSessionManager implements IUserSessionManager {
    */
   makeRequestToSetCookie(
     encryptedCookieData: EncryptedCookieData[],
-    callback: HTTPCallbackFunction,
+    callback: AsyncRequestCallback<any>,
   ) {
+    const currentPageUrl: string = getCurrentPageUrl();
```

Comment:
The URL of page can be anything depending on the content the user is viewing. We need to make a request to the `<scheme><TLD>/<endpoint>`.

For example,
if the page URL is,
https://sub.example.com/some/page/iam/viewing.html

the request should go to,
https://example.com/rsaRequest.

if the page URL is,
https://www.example.com/some/page/iam/viewing.html

the request should go to,
https://example.com/rsaRequest.

if the page URL is,
https://example.com/some/page/iam/viewing.html

the request should go to,
https://example.com/rsaRequest.

Reply:
We should determine the final URL to make a request way earlier than this and store in the state.

Reply:
Addressed